src/Microsoft.ML.CodeGen/Microsoft.ML.CodeGenerator.csproj

```diff
@@ -0,0 +1,114 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <LangVersion>7.3</LangVersion>
+    <PackageId>Microsoft.ML.CodeGenerator</PackageId>
+    <RootNamespace>Microsoft.ML.CodeGenerator</RootNamespace>
+    <AssemblyName>Microsoft.ML.CodeGenerator</AssemblyName>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.CodeAnalysis" Version="2.10.0" />
+    <PackageReference Include="NLog" Version="4.5.11" />
+    <PackageReference Include="Microsoft.ApplicationInsights" Version="2.9.1" />
+    <PackageReference Include="Microsoft.DotNet.PlatformAbstractions" Version="2.1.0" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Microsoft.ML.AutoML\Microsoft.ML.AutoML.csproj" />
+  </ItemGroup>
+  
+  <ItemGroup>
+    <None Update="Templates\Console\Annotation.tt">
+      <Generator>TextTemplatingFileGenerator</Generator>
+      <LastGenOutput>Annotation.txt</LastGenOutput>
+    </None>
+    <None Update="Templates\Console\Annotation.txt">
+      <DesignTime>True</DesignTime>
+      <AutoGen>True</AutoGen>
+      <DependentUpon>Annotation.tt</DependentUpon>
+    </None>
+    <None Update="Templates\Console\ConsumeModel.tt">
+      <Generator>TextTemplatingFilePreprocessor</Generator>
+      <LastGenOutput>ConsumeModel.cs</LastGenOutput>
+    </None>
+    <None Update="Templates\Console\ModelBuilder.tt">
+      <Generator>TextTemplatingFilePreprocessor</Generator>
+      <LastGenOutput>ModelBuilder.cs</LastGenOutput>
+    </None>
+    <None Update="Templates\Console\ModelInputClass.tt">
+      <Generator>TextTemplatingFilePreprocessor</Generator>
+      <LastGenOutput>ModelInputClass.cs</LastGenOutput>
+    </None>
+    <None Update="Templates\Console\ModelOutputClass.tt">
+      <Generator>TextTemplatingFilePreprocessor</Generator>
+      <LastGenOutput>ModelOutputClass.cs</LastGenOutput>
+    </None>
+    <None Update="Templates\Console\ModelProject.tt">
+      <Generator>TextTemplatingFilePreprocessor</Generator>
+      <LastGenOutput>ModelProject.cs</LastGenOutput>
+    </None>
+    <None Update="Templates\Console\PredictProgram.tt">
+      <Generator>TextTemplatingFilePreprocessor</Generator>
+      <LastGenOutput>PredictProgram.cs</LastGenOutput>
+    </None>
+    <None Update="Templates\Console\PredictProject.tt">
+      <Generator>TextTemplatingFilePreprocessor</Generator>
+      <LastGenOutput>PredictProject.cs</LastGenOutput>
+    </None>
+  </ItemGroup>
+
+  <ItemGroup>
```

Comment:
Is this needed?  Look to me project would build without it, I suspect it's something that VS puts there by default.

Reply:
You could also ask Eric Erhardt about this.

Reply:
It's auto-added when you add a text template file into project... So I guess it's just a method to inform Visual Studio that this project has tt file, so Visual Studio can treat the generated CSharp file differently ( see the UI in file explorer for example). 

In that case I guess we can remove that item in mlnet.csproj, cause it doesn't include any tt template any more

Reply:
@eerhardt Please correct me if I'm wrong

Reply:
Unfortunately, yes when you use `.tt` files, this Service will keep being added to your project, even if you try to remove it.

---
In reply to: [315343898](https://github.com/dotnet/machinelearning/pull/4043#discussion_r315343898) [](ancestors = 315343898)